server/core/src/rodsServer.cpp

```diff
@@ -1385,6 +1385,31 @@ int initServerMain(
 
     const auto start_delay_server = [&](std::any&)
     {
+
+
+/*  NEW 4.3.0 LEADER/SUCCESSOR ALGORITHM
+
+   - if self is leader
+     - if successor is defined
+        - if necessary, gracefully complete current jobs and exit
```

Comment:
upon exit, update pid information (remove it) that will go out as response to `irods-grid status` request.